### DISCUSSION TITLE
Adding `{behavior: "smooth"}` option 🏎

### DISCUSSION THREAD

**Discussion by sanjaiyan-dev** [vercel/next.js#39341] [Ideas]
### Describe the feature you'd like to request

Hi,
Adding `{behavior: "smooth"}` to `.scrollIntoView()` function for nicer UI animation/change :)

Ref-: https://github.com/vercel/next.js/pull/39333#pullrequestreview-1062519541

### Describe the solution you'd like

I thought by adding `{behavior: "smooth"}` inside `.scrollIntoView()` would have sleek user experience.

sample-:
```js
 scrollToHash() {
    let { hash } = location
    hash = hash && hash.substring(1)
    if (!hash) return

    const el: HTMLElement | null = document.getElementById(hash)
    if (!el) return

    // If we call scrollIntoView() in here without a setTimeout
    // it won't scroll properly.
    setTimeout(() => el.scrollIntoView({ behavior: "smooth" }), 0)
  }
```

### Describe alternatives you've considered

___

**Comment by ijjk** [comment]
This can be undesirable for some applications that have custom transitions and also during history navigations as you don't want the page scrolling up and down non-stop for smooth behavior. We've also been investigating the opposite and disabling smooth scrolling during a transition if manually configured.
